packages/berry-pnpify/sources/NodePathResolver.ts

```diff
@@ -78,8 +81,32 @@ export class NodePathResolver {
     this.options = options;
   }
 
-  private getIssuer(pnp: PnpApi, pathname: string): string | undefined {
-    const locator = pnp.findPackageLocator(pathname + '/');
+  /**
+   * Returns `readdir`-like result for partially resolved pnp path
+   *
+   * @param issuerInfo issuer package information
+   * @param request either '' or '@scope'
+   *
+   * @returns `undefined` - if dir does not exist, or `readdir`-like list of subdirs in the virtual dir
+   */
+  public readDir(issuerInfo: PackageInformation, request: string): string[] | undefined {
+    const result = [];
+    for (const key of issuerInfo.packageDependencies.keys()) {
+      const [ scope, pkgName ] = key.split('/');
+      if (!request) {
+        if (result.indexOf(scope) < 0) {
+          result.push(scope);
+        }
```

Comment:
Make `result` a `Set` and at the time you return it call `Array.from` on it to build the array. This way if someone work on this function later there are less chances to make a mistake (plus it might be more optimized than calling `indexOf` multiple times).